### ISSUE TITLE
ETELEGRAM: 409 Conflict: terminated by other long poll or webhook

### ISSUE THREAD

**Issue by bennyn** [yagop/node-telegram-bot-api#488]
> Question

I have Telegram Bot running as part of a Node.js script. When terminating the Node.js script by hitting `Ctrl + C` on my keyboard and restarting the script, I am getting the following error from the bot:

> {"code":"ETELEGRAM","message":"ETELEGRAM: 409 Conflict: terminated by other long poll or webhook"}

I guess the error comes from an "unclean" termination of the process (when hitting `Ctrl + C` on my keyboard). What should I do if there is such an interrupt? Do I need to close a session or anything like that?

**Comment by DzianisMakeichyk** [comment]
Reinstall node_modules.
For me its work

**Comment by tryshchenko** [comment]
I use it with https://github.com/remy/nodemon and https://github.com/foreverjs/forever - haven't had such a problem. Several bots are running permanently ~5 months on long polling - very stable. Could you provide more information, please?

**Comment by droidmanspace** [comment]
I am also facing this issue, only changing API token fixes me.

**Comment by mmarinm** [comment]
I am having same issue

**Comment by agitaev** [comment]
I'm having the same issue while running the bot with https://github.com/remy/nodemon

**Comment by droidmanspace** [comment]
@agitaevHi, can u please mention which operating system are you using ???

**Comment by agitaev** [comment]
@droidmanspace hey, I am on macOS 10.13.2. I already solved this issue: my bot was deployed to Heroku and when I launched the bot locally with nodemon, it conflicted - sending the response both, from localhost and Heroku. So, I just turned off the Heroku, while testing locally.

I'm not sure is it the case for others or not

**Comment by ricustr** [comment]
Hi, I have the same error in getUpdates method:
"Error: ETELEGRAM: 409 Conflict: terminated by other getUpdates request; make sure that only one bot instance is running", does somebody know more information or how to check running instances? Thanks.

**Comment by gyoza** [comment]
I am getting this with a new hubot i am trying to run... is hubot still supported using the telegram adapter?

**Comment by sabino** [comment]
@gyoza I also getting this error

**Comment by snics** [comment]
I have the same error.

**Comment by HOllarves** [comment]
Having the same problem. I even restarted my computer to make sure no process got stuck closing. Still just using it in development, so the only instance always is the one I serve in my local environment. Hope someone could point me in the right direction.

Thanks!

**Comment by thevafelkoman** [comment]
I just recreate Bot Token and the error is gone , i don't know is it works for all of you, but for me it works

**Comment by andreafspeziale** [comment]
I still have the same error even revoking the current API key.

Removing the API key u get this error:
`error: [polling_error] {"code":"ETELEGRAM","message":"ETELEGRAM: 401 Unauthorized"}`

Adding the new API key:
`error: [polling_error] {"code":"ETELEGRAM","message":"ETELEGRAM: 409 Conflict: terminated by other getUpdates request; make sure that only one bot instance is running"}`

Still no ideas?

Thanks all.

**Comment by whitecolor** [comment]
Try to increase the polling interval
```
polling: {
   interval: 1000
}
```

**Comment by alihesari** [comment]
@andreafspeziale I also have your problem. Have you solved the problem?

**Comment by andreafspeziale** [comment]
@alihesari I've stop working on my bots since a while. So I don't really know. But I've never fixed it.

**Comment by rotimi-best** [comment]
> Try to increase the polling interval
> 
> ```
> polling: {
>    interval: 1000
> }
> ```

It didn't solve the problem. I also still have issue. I also tried revoking, that fixed the issue for a while then after some time I kept on seeing that error

`[polling_error] {"code":"ETELEGRAM","message":"ETELEGRAM: 409 Conflict: terminated by other getUpdates request; make sure that only one bot instance is running"}`

**Comment by serwer20** [comment]
same here but i'm working from firebase functions

**Comment by Sk-1599** [comment]
> I just recreate Bot Token and the error is gone , i don't know is it works for all of you, but for me it works

Yes it worked for first time but it is still giving error when we terminate it conflict error

**Comment by markpumpkin** [comment]
> @droidmanspace hey, I am on macOS 10.13.2. I already solved this issue: my bot was deployed to Heroku and when I launched the bot locally with nodemon, it conflicted - sending the response both, from localhost and Heroku. So, I just turned off the Heroku, while testing locally.
> 
> I'm not sure is it the case for others or not

Thanks, it worked for me

**Comment by bennycode** [comment]
What also helps is to implement a graceful shutdown of the Telegram bot.

**Example:**

```ts
process.once("SIGINT", () => bot.stop());
process.once("SIGTERM", () => bot.stop());
``` 

I also had less complications by switching to the [telegraf.js](https://www.npmjs.com/package/telegraf) library.
